src/Mastodon.elm

```diff
@@ -603,3 +636,70 @@ unfavourite client id =
     HttpBuilder.post (client.server ++ "/api/v1/statuses/" ++ (toString id) ++ "/unfavourite")
         |> HttpBuilder.withHeader "Authorization" ("Bearer " ++ client.token)
         |> HttpBuilder.withExpect (Http.expectJson statusDecoder)
+
+
+subscribeToWebSockets : Client -> StreamType -> (String -> a) -> Sub a
+subscribeToWebSockets client streamType message =
+    let
+        type_ =
+            case streamType of
+                UserStream ->
+                    "user"
+
+                LocalPublicStream ->
+                    "public:local"
+
+                GlobalPublicStream ->
+                    "public:local"
+
+        url =
+            (Util.replace "http" "ws" client.server)
```

Comment:
Nit: Reading this code I thought we were using `ws://` instead of `wss://`. We should probably write `Util.replace "https" "wss" client.server` (`https` is a hard constraint for mastodon API root url).